bundles/org.openhab.core/src/main/java/org/openhab/core/util/Statistics.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.util;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link Statistics} is a class with statistics helper methods.
+ *
+ * @author Mark Herwege - Initial contribution
+ */
+@NonNullByDefault
+public class Statistics {
+
+    public static boolean randomQuickSelectSeed = false; // Can be enabled to always create a random pivot index for the
```

Comment:
What would be the benefit if that? I can't imagine a use-case for where random pivots would be beneficial.

Reply:
It was easy for testing the algorithm, and allowed to stay close to the textbook algorithm. Removed.